core/src/test/scala/org/apache/spark/deploy/client/AppClientSuite.scala

```diff
@@ -122,7 +134,9 @@ class AppClientSuite extends SparkFunSuite with LocalSparkContext with BeforeAnd
     val ci = new AppClientInst(masterRpcEnv.address.toSparkURL)
 
     // requests to master should fail immediately
-    assert(ci.client.requestTotalExecutors(3) === false)
+    whenReady(ci.client.requestTotalExecutors(3), timeout(0.seconds)) { success =>
```

Comment:
nit: don't use `0` timeout. It depends on the implementation and it could check the timeout at first before running the command in future.